app/components/elements/form/error/index.js

```diff
@@ -0,0 +1,12 @@
+import {fieldState} from '../selectors'
+import BaseError from './base'
+
+export const Error = ({formName, name, ...props}) => {
+  const {error, asyncError, focus, touched} = fieldState(formName, name)
+  return BaseError({
+    error: touched && !focus && (error || asyncError),
```

Comment:
What if I want to show an error when the field is focused? Or show an error from API?

Also not clear in the code what the difference between asyncError and error is.. Doesn't seem like they get any varied treatment. Why not just error?

Reply:
Error from API is an asyncError. The onSubmit callback that's passed to form component has three args data, onSuccess, and onError. onError can be called with onError({fieldErrors}) (which maybe should be called asyncErrors instead). Will have to rethink if both asyncErrors and errors are necessary.

Reply:
I think it's because form error validations are run every time form rerenders, and running those validations will never result in the same errors as asyncErrors so the asyncErrors would always get removed.